presto-main/src/main/java/com/facebook/presto/server/security/LdapAuthenticator.java

```diff
@@ -188,7 +190,32 @@ private static String decodeCredentials(String base64EncodedCredentials)
     private Principal authenticate(String user, String password)
             throws AuthenticationException
     {
-        Map<String, String> environment = createEnvironment(user, password);
+        AuthenticationException ex = null;
```

Comment:
As I understand, and based on reading [this](https://docs.oracle.com/javase/tutorial/jndi/ldap/exceptions.html), there are two types of LDAP errors:

* `javax.naming.AuthenticationException`: Invalid credentials. Prompt the user again for credentials.
* Any other `NamingException`: There was an error talking to LDAP, or the LDAP server doesn't like us, etc., so having valid credentials (which we might have) won't help, no need to prompt the user.

We should consider how that should be handled here. If the first call fails due to invalid credentials, and the second one fails due to another error (maybe transient), should we prompt the user? Also, what error should we return? All of them? The last one? Combine together with de-duplication? Or maybe we shouldn't return the error from the LDAP server (not sure that is a good idea anyway).

Note that we prompt the user by setting an HTTP authenticate header value when throwing `AuthenticationException`. For the Presto CLI and JDBC driver, we always pre-authenticate, so prompting is ignored. This is recommend for performance reasons (avoids a round-trip), and it's possible most/all clients do this, so the prompting part might not matter.

This is more relevant with #7632 which splits out LDAP with an API that's decoupled from HTTP.